pkg/apis/infinispan/v1/types_util.go

```diff
@@ -155,6 +155,11 @@ func (ispn *Infinispan) ApplyDefaults() {
 	if ispn.Spec.Security.EndpointAuthentication == nil {
 		ispn.Spec.Security.EndpointAuthentication = pointer.BoolPtr(true)
 	}
+	if *ispn.Spec.Security.EndpointAuthentication == true {
+		if ispn.Spec.Security.EndpointSecretName == "" {
+			ispn.Spec.Security.EndpointSecretName = ispn.GetSecretName()
```

Comment:
`GetSecretName` already validates empty value inside. It seems to be a simple equals instead of comparision

Reply:
yeah thanks @dmvolod !